### PR TITLE
Bugfix: hx-boost + hx-history-elt working on same page

### PR DESCRIPTION
**Pull request by HeyHugo**
Proposed fix for #1660

This fixes issues with browser back button when boosted links and hx-history-elt are on the same page. Whenever hx-boost adds a record to the history cache we store the property `boosted: true` along with the record. That way we can know if we should target the body when restoring from cache.

### CODE REVIEW

File: test/index.html [hunk 1]
```diff
@@ -23,7 +23,7 @@ <h2>Scratch Page</h2>
 </ul>
 
 <h2>Manual Tests</h2>
-<a href="manual">Here</a>
+<a href="manual/">Here</a>
```
Comment (alexpetros) [inline test/index.html:26]: Why is this necessary?

**Comment by alexpetros** [comment]
Don't use these features myself so sending this one up to @1cg

File: test/index.html (hunk 1 above)
Comment (HeyHugo) [reply test/index.html:26]: The link to manual tests was broken on the `/test` page without trailing slash

I now noticed this is a problem within more sub-directories, e.g all the history subdirectories under `manual` shuld probably have trailing slash as well.

**Comment by HeyHugo** [comment]
Added a commit to fix `undefined` being set as argument in `saveCurrentPageToHistory()` so we only ever set `boosted` to `true`/`false`

**Comment by croxton** [comment]
History restores should inject cached content into the `hx-history-elt` element whether from a boosted link or not. That's the intended behaviour, not a bug. 

By default htmx targets `<body>` if `hx-history-elt` isn't defined,  but I usually use `<main>`. This PR would seem to force all history restores from boosted links to snapshot and restore into `<body>` which would break that implementation.

**Comment by HeyHugo** [comment]
@croxton Hmm I see now what you mean, yes that would break things. I probably didn't use `hx-history-elt` like it was intended really. I used `hx-history-elt` on a modal type of element where back button had to close the modal. There were hx-boosted links to pages unrelated with the modal which had their back navigation broken.

`hx-history-elt` is a all pages or none kind of feature since it leaks into other pages by requiring them to also have `hx-history-elt` to not break hx-boost + back navigation.

This PR can be closed as I see it'd break backwards compatibility. I still think my idea in the sidenote of #1660 regarding `hx-history-target` would be really useful in cases like mine. Perhaps something for a future extension or htmx v2 :)

**Comment by croxton** [comment]
@HeyHugo a pattern I often use it to put a container for a modal _outside_ of the `hx-history-elt` element and swap content into it and pushing the url so it creates a history entry. Then I ensure that the url loads as a standalone page showing the modal content. This works with the browser's forward/backwards buttons just like visiting a normal page. Here's a real example: https://level.law/team

In terms of managing areas of the content saved in the history cache, that is also possible using the `htmx:historyItemCreated` event if you have a specific requirement. This extension should give you some ideas:
https://gist.github.com/croxton/e2c33bd22591f9a5bd8c9d23a56c9edc
